plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -68,10 +69,11 @@ public class IcebergTableHandle
     private final boolean recordScannedFiles;
     private final Optional<DataSize> maxScannedFileSize;
 
-    // ANALYZE only
-    private final Optional<Set<String>> analyzeColumns;
+    // ANALYZE only. Coordinator-only
+    private final Optional<Boolean> forAnalyze;
```

Comment:
Do we need the Optional? Can it be false by default, and then set to true only by the `forAnalyze()` method?

Reply:
yes, technically it could.  Optional guarantees that we either know the truth ("is it for Analyze?") or we fail.